packages/webhid/src/methods.ts

```diff
@@ -69,13 +69,28 @@ export async function setupXkeysPanel(browserDevice: HIDDevice): Promise<XKeys>
 		})
 	})
 
-	// Wait for the device to initialize:
-	try {
-		await xkeys.init()
-
-		return xkeys
-	} catch (e) {
-		await deviceWrap.close()
-		throw e
-	}
+	let alreadyRejected = false
+	return new Promise((resolve, reject) => {
+		const xkeysStopgapErrorHandler = (e: unknown) => {
+			if (alreadyRejected) {
+				console.error(`Xkeys: Error emitted after setup already rejected:`, e)
+				return
+			}
+			deviceWrap.close()
+				.then(() => reject())
```

Comment:
Minor note: is there a reason the `reject()` has no argument here? Should we `reject(e)`?

Reply:
Good catch, will make this change